megazords/ios-rust/README.md

```diff
@@ -0,0 +1,108 @@
+      architectures and their subdirectories.
+* The `build-xcframework.sh` script which knows how to stitch things together into a full
+  XCFramework bundle.
+    * The XCFramework format is not well documented; briefly:
+        * It's a directory containing resources compiled for multiple target architectures,
+          typically distributed as `.zip` file.
+        * The top-level directory contains a subdirectory per architecture, and an `Info.plist`
+          file that says what things live in which directory.
+        * Each subdirectory contains a `.framework` directory for that architecture. There
+          are notes on the layout of an individual `.framework` in the links below.
+
+It's a little unusual that we're building the XCFramework by hand, rather than defining it
+as the build output of an Xcode project. It turns out to be simpler for our purposes, but
+does risk diverging from the expected format if Apple changes the detailts of XCFrameworks
+in future Xcode releases.
+
+## Adding crates
+
+To add a new crate to the distribution:
+
+1. Update its `uniffi.toml`, if any, to include the following settings:
+    ```
+    [bindings.swift]
+    ffi_module_name = "MozillaRustComponents"
+    ffi_module_filename = "<crate_name>FFI"
+    ```
+1. Add it as a dependency in `Cargo.toml`.
+1. Add a `pub use` declaration for it in `./src/lib.rs`.
+1. Add logic to `build-xcframework.sh` to copy or generate its header file into the build.
+1. Add a `#import` for its header file to `MozillaRustComponents.h`
+
+## Testing locally
+
+For release builds, we publish the resulting `MozillaRustComponents.xcframework.zip` as a GitHub
+release artifact, and then update https://github.com/mozilla/rust-components-swift to point to
+it via URL and hash.
+
+For testing from a PR or unreleased git commit, you can:
+
+* Find the CircleCI job named `ios-artifacts` for the commit you want to test, click through to view it on CircleCI, and confirm that it completed successfully.
+* In the "artifacts" list, locate `MozillaRustComponents.xcframework.zip` and note its URL.
+* In the "steps" list, find the step named `XCFramework bundle checksum`, and note the checksum in its output.
+* Take a local checkout of https://github.com/mozilla/rust-components-swift,
+and edit its `Swift.package` to use the above URL and checksum for the `MozillaRustComponents` binary target.
+* Commit the result to your local checkout, and make a git tag in `MAJOR.MINOR.PATCH` format.
+* Add your local `rust-components-swift` repo as a Swift Package dependency in a consuming app,
+  by specifying `file://path/to/rust-components-swift` as the git repo.
+
+Note that your changes *must* be included in a local git tag. Swift Package Manager is very
+opinionated and will only pull dependencies from a git tag that it can parse as a semver
+version number.
+
+For local testing without relying on CircleCI, you can also:
```

Comment:
A super minor organizational thought (feel free to completely discarding this, I'm just making sure I note down everything I think of)

I read through this section and was first thinking "wait you can't do this locally without a PR?" until I hit this line, could be worth it to change the titles, maybe have this whole  section be a `## Testing` section, and have `### Testing Locally`, `### Testing a PR` and `### Testing a release` as separate sections - again for the sake of time, we can do this later ourselves 😄

Reply:
Good idea, thanks.